test/test_sessionmanager.py

```diff
@@ -0,0 +1,105 @@
+import unittest
+
+from urllib3.sessionmanager import SessionManager
+from urllib3.poolmanager import ProxyManager, PoolManager
+from urllib3.util.sessioncontext import DefaultCookiePolicy, CookieJar, SessionContext
+from urllib3.packages import six
+from urllib3.response import HTTPResponse
+from urllib3.request import Request
+
+cookielib = six.moves.http_cookiejar
+
+
+class TestSessionManager(unittest.TestCase):
+
+    def setUp(self):
+        self.manager = SessionManager(PoolManager())
+
+    def test_set_cookie_policy(self):
+        policy = cookielib.DefaultCookiePolicy()
+        self.assertNotEqual(policy, self.manager.context.cookie_jar._policy)
+        self.manager.context.cookie_jar.set_policy(policy)
```

Comment:
I initially had a helper method at the `SessionManager` level to handle that; we had moved away from it as it too tightly coupled `SessionManager` to the lower levels. I do have a draft concept that may work here and maintain generic-ness, but I'd need to bring in all the commits from a later 
branch to provide the restructuring necessary. Would you prefer that on this PR, or would you prefer to merge this, and then see that draft?